packages/helper-grammar-regex-collection/index.js

```diff
@@ -1,15 +1,81 @@
 import go from './go.js';
+import regexBuilder from './regexBuilder';
 
-const REQUIRE = /require(?:\.resolve)?(?:\s|\()\s*['"]([^'"\s]+)['"]\s*\)?/g;
-const IMPORT = /import [\r\n\s\w{},*\$]*(?: from )?['"]([^'"\s]+)['"]/g;
-const EXPORT = /export [\r\n\s\w{},*\$]*(?: from )['"]([^'"\s]+)['"]/g;
-const GEM = /gem ['"]([^'"\s]+)['"]/g;
-const HOMEBREW = /(?:depends_on|conflicts_with)(?: cask:| formula:)? ['"]([^'"\s]+)['"]/g;
-const TYPESCRIPT_REFERENCE = /\/{3}\s?<reference path=['"]([^'"\s]+)['"]/g;
-const DOCKER_FROM = /FROM\s([^\n]*)/g;
-const VIM_PLUGIN = /(?:(?:(?:Neo)?Bundle(?:Lazy|Fetch)?)|Plug(?:in)?)\s['"]([^'"\s]+)['"]/g;
-const RUST_CRATE = /(?:extern crate|use) ([^:; ]+)/g;
-const PYTHON_IMPORT = /^\s*(?:import|from)\s([^\s]*)/gm;
+const subpatterns = {};
+const regex = regexBuilder(subpatterns);
```

Comment:
Yeah, the naming is a little strange here. My perspective on it is that `regex` is a function that takes a string and returns a RegExp, much like its constructor: `RegExp('my pattern')`, so the `regex` name makes sense to me.

However, `regexBuilder` is still a little awkward. While it does build the `regex` function, it still sounds like it might do the same thing as `regex`. Besides, I discovered there's already another `regexBuilder` in the codebase ([lib/plugins/helper/regex-builder.js](https://github.com/OctoLinker/browser-extension/blob/cc1bae8b95d60d14c058ce1e610e84619395bfeb/lib/plugins/helper/regex-builder.js)). I not sure what a better name would be, though. Some ideas:

* `regexTagBuilder`
* `regexTagFunctionBuilder`